main.cpp

```diff
@@ -0,0 +1,320 @@
+        point1 = cv::Point(0, b - distance);
+        point2 = cv::Point(500, b + distance);
+    } else {
+        point1 = cv::Point(0, b - distance * k);
+        point2 = cv::Point(500, (b + distance) * k);
+    }
+    line_1 = get_line(point1, point2);
+    line_2 = get_line(point2, point1);
+}
+
+bool is_in_area(cv::Point point, cv::Vec4f line1, cv::Vec4f line2) {
+    float x = point.x, y = point.y;
+    float a1 = line1[1] / line1[0], b1 = line1[3] / line1[0];
+    float a2 = line2[1] / line2[0], b2 = line2[3] / line2[0];
+    if (a1 > 9999 || a1 < -9999) {
+        float value2 = y - a2 * x - b2;
+        if (value2 >= 0) {
+            return true;
+        } else {
+            return false;
+        }
+    } else if (a2 > 9999 || a2 < -9999) {
+        float value1 = y - a1 * x - b1;
+        if (value1 >= 0) {
+            return true;
+        } else {
+            return false;
+        }
+    } else {
+        float value1 = y - a1 * x - b1;
+        float value2 = y - a2 * x - b2;
+        if (value1 * value2 >= 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+}
+
+int nearest_multiple(int x) {
+    int n = floor((x - 40) / 100);
+    return 50 + 100 * n;
+}
+
+
+void add_to_vector_if_not_exists(std::vector<std::pair<cv::Point, cv::Point>> &lines,
+                                 const std::pair<cv::Point, cv::Point> &element) {
+    for (const auto &line: lines) {
+        if (line == element) {
+            return;
+        }
+    }
+    lines.push_back(element);
+}
+
+void compare_and_swap(int &x_0, int &y_0, int &x_1, int &y_1, bool flag) {
+    if (flag == 0) {
+        if (y_0 < y_1) {
+            swap(y_0, y_1);
+            swap(x_0, x_1);
+        }
+    } else {
+        if (x_0 < x_1) {
+            swap(y_0, y_1);
+            swap(x_0, x_1);
+        }
+    }
+}
+
+float get_length(int &x_0, int &y_0, int &x_1, int &y_1, int flag) {
+    if (flag == 0) {
+        return (y_1 - y_0);
+    } else if (flag == 1) {
+        return (x_1 - x_0);
+    } else {
+        return sqrt((x_1 - x_0) * (x_1 - x_0) + (y_1 - y_0) * (y_1 - y_0));
+    }
+}
+
+int matrix_size, num_lines;
+vector<int> path;
+vector<vector<int>> edges;
+vector<vector<bool>> visited_edge;
+vector<int> target_path;
+
+bool backtrack(int u) {
+    if (path.size() == num_lines + 1) {
+        target_path = path;
+        return true;
+    }
+    bool flag = false;
+    for (int i = 0; i < matrix_size; i++) {
+        if (edges[u][i] && !visited_edge[u][i] && !visited_edge[i][u]) {
+            path.push_back(i);
+            visited_edge[u][i] = true;
+            visited_edge[i][u] = true;
+            if (backtrack(i)) flag = true;
+            visited_edge[u][i] = false;
+            visited_edge[i][u] = false;
+            path.pop_back();
+        }
+    }
+    return flag;
+}
+
+
+int main() {
+    char img_path[80];
+    cin >> img_path;
+    Mat img = cv::imread(img_path);
+    Mat img_COPY = img;
```

Comment:
错误处理，mat直接赋值为浅拷贝，需要复制数据应使用
```suggestion
    Mat img_COPY = img.clone();
```